### PR TITLE
Add libEnsemble

### PR DESCRIPTION
**Pull request by ax3l**
Add a recipe for libEnsemble.

Repo: https://github.com/Libensemble/libensemble
Docs: https://libensemble.readthedocs.io
License: BSD-3-Clause

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] ~~If static libraries are linked in, the license of the static library is packaged.~~
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/libensemble/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    can benefit from increased concurrency levels.
+    libEnsemble aims for: Extreme scaling, Resilience/fault tolerance,
+    Monitoring/killing jobs (and recovering resources), Portability and
+    flexibility, Exploitation of persistent data/control flow.
+  dev_url: https://github.com/Libensemble/libensemble
+
+extra:
+  recipe-maintainers:
+    - ax3l
+    - shuds13
```
Comment (ax3l) [inline recipes/libensemble/meta.yaml:58]: cc @shuds13: I'll give more details in our meeting later today. who else want's to co-maintain this automated feedstock (by github handle)?

Please comment with "please add me" here.

Comment (jlnav) [reply recipes/libensemble/meta.yaml:57]: Please add jlnav

**Comment by jmlarson1** [comment]
Looks good

File: recipes/libensemble/meta.yaml (hunk 1 above)
Comment (shuds13) [reply recipes/libensemble/meta.yaml:57]: I am happy to be listed as a package maintainer.

Comment (jmlarson1) [reply recipes/libensemble/meta.yaml:57]: Please add me

Comment (isuruf) [reply recipes/libensemble/meta.yaml:58]: @ax3l, was it intentional that your name isn't here?

Comment (ax3l) [reply recipes/libensemble/meta.yaml:58]: Yep, I am a downstream consumer :)

Comment (ax3l) [reply recipes/libensemble/meta.yaml:58]: @isuruf ready for merge :)

**Pull request merged**